flink/v1.19/flink/src/test/java/org/apache/iceberg/flink/sink/TestCompressionSettings.java

```diff
@@ -91,19 +94,36 @@ public void testCompressionAvro() throws Exception {
             ImmutableMap.of(FlinkWriteOptions.WRITE_FORMAT.key(), "AVRO"));
 
     if (initProperties.get(TableProperties.AVRO_COMPRESSION) == null) {
-      Assert.assertEquals(
-          TableProperties.AVRO_COMPRESSION_DEFAULT,
-          resultProperties.get(TableProperties.AVRO_COMPRESSION));
-      Assert.assertEquals(
-          TableProperties.AVRO_COMPRESSION_LEVEL_DEFAULT,
-          resultProperties.get(TableProperties.AVRO_COMPRESSION_LEVEL));
+      assertThat(resultProperties)
+          .containsEntry(TableProperties.AVRO_COMPRESSION, TableProperties.AVRO_COMPRESSION_DEFAULT)
+          .satisfies(
+              props -> {
+                if (props.containsKey(TableProperties.AVRO_COMPRESSION_LEVEL_DEFAULT)) {
+                  assertThat(props)
+                      .containsEntry(
+                          TableProperties.AVRO_COMPRESSION_LEVEL_DEFAULT,
+                          TableProperties.AVRO_COMPRESSION_LEVEL);
+                } else {
+                  assertThat(props)
+                      .doesNotContainKey(TableProperties.AVRO_COMPRESSION_LEVEL_DEFAULT);
+                }
+              });
     } else {
-      Assert.assertEquals(
-          initProperties.get(TableProperties.AVRO_COMPRESSION),
-          resultProperties.get(TableProperties.AVRO_COMPRESSION));
-      Assert.assertEquals(
-          initProperties.get(TableProperties.AVRO_COMPRESSION_LEVEL),
-          resultProperties.get(TableProperties.AVRO_COMPRESSION_LEVEL));
+      assertThat(resultProperties)
+          .containsEntry(
+              TableProperties.AVRO_COMPRESSION,
+              initProperties.get(TableProperties.AVRO_COMPRESSION))
+          .satisfies(
+              props -> {
+                if (props.containsKey(TableProperties.AVRO_COMPRESSION_LEVEL)) {
```

Comment:
why is the if/else needed here and not in the original assertion?

Reply:
Sorry this is actually not needed. Will push the fix.